File: app/src/main/java/fr/gstraymond/android/HistoryActivity.java
package fr.gstraymond.android;

import android.app.AlertDialog;
import android.content.DialogInterface;
import android.os.Bundle;
import android.support.v7.widget.Toolbar;
import android.view.Menu;
import android.view.MenuInflater;
import android.view.MenuItem;

import java.util.ArrayList;

import fr.gstraymond.R;
import fr.gstraymond.android.fragment.HistoryListFragment;
import fr.gstraymond.db.json.JsonHistory;

import static fr.gstraymond.constants.Consts.HISTORY_LIST;

public class HistoryActivity extends CustomActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_history);

        Toolbar toolbar = (Toolbar) findViewById(R.id.toolbar);
        setSupportActionBar(toolbar);
        getSupportActionBar().setDisplayHomeAsUpEnabled(true);

        showHistory();

        actionBarSetDisplayHomeAsUpEnabled(true);
    }

    @Override
    public boolean onOptionsItemSelected(MenuItem item) {
        switch (item.getItemId()) {
            case android.R.id.home:
                finish();
                return true;

            case R.id.history_clear_tab:
                AlertDialog.Builder alert = new AlertDialog.Builder(this);
                alert.setTitle(getString(R.string.history_alert_title));
                alert.setMessage(getString(R.string.history_alert_description));

                alert.setPositiveButton(getString(R.string.history_alert_ok), new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int whichButton) {
                        getJsonHistoryDataSource().clearNonFavoriteHistory();
                        showHistory();
                    }
                });
                alert.setNegativeButton(getString(R.string.history_alert_cancel), new DialogInterface.OnClickListener() {
                    public void onClick(DialogInterface dialog, int whichButton) {
                        //So sth here when "cancel" clicked.
                    }
                });
                alert.show();
                return true;
        }

        return super.onOptionsItemSelected(item);
    }

    @Override
    public boolean onCreateOptionsMenu(Menu menu) {
        MenuInflater inflater = getMenuInflater();
        inflater.inflate(R.menu.card_history_menu, menu);
        return true;
    }

    private void showHistory() {
        ArrayList<JsonHistory> allHistory = getJsonHistoryDataSource().getAllHistory();

        Bundle bundle = new Bundle();
        bundle.putParcelableArrayList(HISTORY_LIST, allHistory);

        replaceFragment(new HistoryListFragment(), R.id.history_fragment, bundle);
    }
}

File: app/src/main/kotlin/fr/gstraymond/android/HistoryActivity.kt
package fr.gstraymond.android

import android.app.AlertDialog
import android.content.Context
import android.content.Intent
import android.os.Bundle
import android.view.Menu
import android.view.MenuItem
import fr.gstraymond.R
import fr.gstraymond.android.fragment.HistoryListFragment
import fr.gstraymond.constants.Consts.HISTORY_LIST
import fr.gstraymond.models.Board
import fr.gstraymond.models.Board.DECK
import fr.gstraymond.utils.find
import java.util.*

class HistoryActivity : CustomActivity(R.layout.activity_history) {

    companion object {
        val DECK_ID_EXTRA = "deck_id"
        val BOARD_EXTRA = "board"

        fun getIntent(context: Context,
                      deckId: String? = null,
                      board: Board = DECK) =
                Intent(context, HistoryActivity::class.java).apply {
                    putExtra(DECK_ID_EXTRA, deckId)
                    putExtra(BOARD_EXTRA, board.ordinal)
                }
    }

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        setSupportActionBar(find(R.id.toolbar))
        true.actionBarSetDisplayHomeAsUpEnabled()

        showHistory()
    }

    override fun onOptionsItemSelected(item: MenuItem) = when (item.itemId) {
        android.R.id.home -> {
            finish()
            true
        }

        R.id.history_clear_tab -> {
            AlertDialog.Builder(this)
                    .setTitle(getString(R.string.history_alert_title))
                    .setMessage(getString(R.string.history_alert_description))
                    .setPositiveButton(getString(R.string.history_alert_ok)) { _, _ ->
                        jsonHistoryDataSource.clearNonFavoriteHistory()
                        showHistory()
                    }
                    .setNegativeButton(getString(R.string.history_alert_cancel)) { _, _ -> }
                    .show()
            true
        }

        else -> super.onOptionsItemSelected(item)
    }

    override fun onCreateOptionsMenu(menu: Menu) =
            menuInflater.inflate(R.menu.card_history_menu, menu).run { true }

    private fun showHistory() {
        val bundle = Bundle().apply {
            val allHistory = ArrayList(jsonHistoryDataSource.all().sortedByDescending { it.date })
            putParcelableArrayList(HISTORY_LIST, allHistory)
            putString(DECK_ID_EXTRA, intent.getStringExtra(DECK_ID_EXTRA))
            putInt(BOARD_EXTRA, intent.getIntExtra(BOARD_EXTRA, 0))
        }

        replaceFragment(HistoryListFragment(), R.id.history_fragment, bundle)
    }
}
